### PR TITLE
launcher: Fix memory leak

### PR DESCRIPTION
**Pull request by rbuj**

### CODE REVIEW

File: mate-panel/launcher.c [hunk 1]
```diff
@@ -677,8 +676,9 @@ panel_launcher_find_writable_uri (const char *launcher_location,
 		return uri;
 	}
 
-	filename = panel_launcher_get_filename (launcher_location);
+	char *filename = panel_launcher_get_filename (launcher_location);
```
Comment (cwendling) [inline mate-panel/launcher.c:679]: style: shouldn't the declaration be separate from the assignation?  Basically, style around seems to suggest the declaration should have been left alone.

Comment (rbuj) [reply mate-panel/launcher.c:679]: two cases in which the declaration and assignment are made at the same line:

https://github.com/mate-desktop/mate-panel/blob/efa9913ab6bce6f8c74517808c561c344291e711/mate-panel/launcher.c#L249

https://github.com/mate-desktop/mate-panel/blob/efa9913ab6bce6f8c74517808c561c344291e711/mate-panel/launcher.c#L294

you can see the repetition of this criterion on #1128

Comment (sc0w) [reply mate-panel/launcher.c:679]: I agree with @cwendling

Comment (rbuj) [reply mate-panel/launcher.c:679]: In such case you should propose to use `-Wdeclaration-after-statement`, with which I disagree.
https://gcc.gnu.org/onlinedocs/gcc-9.1.0/gcc/Mixed-Declarations.html#Mixed-Declarations
```
$ CFLAGS="-Wdeclaration-after-statement" ./autogen.sh --prefix=/usr && make &> make.log
$ grep declaration-after-statement make.log 
eggdesktopfile.c:1195:7: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-applet-frame-dbus.c:381:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-gtk.c:198:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-gtk.c:236:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-launch.c:113:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
main.c:186:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-widget.c:195:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
button-widget.c:568:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
button-widget.c:588:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
applet.c:601:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
applet.c:607:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
drawer.c:315:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-properties-dialog.c:875:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-run-dialog.c:175:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-run-dialog.c:309:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-run-dialog.c:908:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
menu.c:428:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
menu.c:566:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
menu.c:572:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
menu.c:955:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
menu.c:1113:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
menu.c:1271:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-context-menu.c:337:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-context-menu.c:343:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-menu-button.c:383:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-menu-button.c:463:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-menu-items.c:1126:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-toplevel.c:2928:4: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-toplevel.c:4229:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-multimonitor.c:67:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-bindings.c:138:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-layout.c:270:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-layout.c:311:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-profile.c:1180:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
panel-addto.c:644:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
mate-panel-applet.c:849:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
mate-panel-applet.c:855:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
mate-panel-applet.c:859:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
mate-panel-applet.c:1840:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
mate-panel-applet.c:1843:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
clock.c:869:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
clock.c:2411:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
clock.c:3356:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
system-timezone.c:269:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
sn-dbus-menu-item.c:185:11: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
na-tray-child.c:566:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
main.c:312:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
na-grid.c:171:7: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
window-list.c:698:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
workspace-switcher.c:571:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
```

Comment (rbuj) [reply mate-panel/launcher.c:679]: PR #1130 by @cwendling.

https://github.com/mate-desktop/mate-panel/blob/7d706e9fe1f53ef73f644eb1a6488239541d6714/mate-panel/panel.c#L612

https://github.com/mate-desktop/mate-panel/blob/7d706e9fe1f53ef73f644eb1a6488239541d6714/mate-panel/panel.c#L616-L617

@sc0w making cross-references is a bad practice... @cwendling I'm sorry for the inconvenience.

Comment (cwendling) [reply mate-panel/launcher.c:679]: Indeed.  I can fix this if wanted.

However, I also see a difference: personally I like to stick with the declarations at the start of the block, regardless of a declaration/assignation on the same line rule or not.

But ultimately I don't care so much so long as we're consistent.

Comment (rbuj) [reply mate-panel/launcher.c:679]: I usually let the author choose. I agree there are some cases where it's cleaner to use the assignment and declaration on the same line, like using a variable in a small scope or it's released shortly after declaring it. if the block is at a level with a lot of bleeding, it is better to use two lines. In contrast, if the block has a lot of indentation, two lines are recommended. The common sense will prevail in the end, if the author tries to do his job in the best way.

**Comment by raveit65** [comment]
@lukefromdc 
We consider memleak fixes as bugfixes and you can cherry-pick such commits to stable branch after merging to master.
Thx

**Comment by raveit65** [comment]
Done for you https://github.com/mate-desktop/mate-panel/commit/16ffeaf0a87f53b2ebbb679c2899fc472693f421

**Pull request merged**